-1

Tested at: 7f288009712f9c7117bf9a96e03d9247a370dad8


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/09ca96da5ef42a2e5f6970580b1194bd51158cb0
https://github.com/cms-sw/cmssw/commit/52ee4d0d0af1f56d49f7c29c1362d22825a80f13
https://github.com/cms-sw/cmssw/commit/d3aa85678e550cfd1f67138402a9638a50203890
https://github.com/cms-sw/cmssw/commit/6392f47e8d62121391f89e676d3a2635714b1741
https://github.com/cms-sw/cmssw/commit/fcd293dacbb35e50cee14eed39d823f1109ffed0
https://github.com/cms-sw/cmssw/commit/981939966af9100b414f344a153d94725044de41
https://github.com/cms-sw/cmssw/commit/c75f9f30c4fecf484e19874cb9dd7e30d997d554
https://github.com/cms-sw/cmssw/commit/b6eda340bd0a999942117f1d52da449a5ad28b95
https://github.com/cms-sw/cmssw/commit/62344a9088bdf3bcbaeb682005d1f1e7684de15e
https://github.com/cms-sw/cmssw/commit/06d1fe3cfefb629e50e980ebaae63a77b35ed7c7
https://github.com/cms-sw/cmssw/commit/03c43df9885cf604235c7563837928101145f473
https://github.com/cms-sw/cmssw/commit/e97d88ded6968a6c2fbcbcce57e1eaa6777e4c6a
https://github.com/cms-sw/cmssw/commit/c7ced6ae014d190ad85e440c12ff3f44f77323f1
https://github.com/cms-sw/cmssw/commit/66153ad310d01be1d512ceb632728cd0191b310a
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18211/18935/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18211/18935/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18211/18935/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/DQMServices/Core/bin/MergeFile.cpp 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/DQMServices/Core/test/DQMTestStandaloneBuildOfDQMStore.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/DQMServices/Core/test/DQMFastMatchTest.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/DQMServices/Core/test/DQMQualityTestsExample.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/SLHCUpgradeSimulations/Geometry/src/PrintGeomMatInfo.cc: In member function 'void PrintGeomMatInfo::dumpSummary(std::ostream&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/SLHCUpgradeSimulations/Geometry/src/PrintGeomMatInfo.cc:174:11: error: 'G4PhysicalVolumeStore' does not name a type
     const G4PhysicalVolumeStore * pvs = G4PhysicalVolumeStore::GetInstance();
           ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/SLHCUpgradeSimulations/Geometry/src/PrintGeomMatInfo.cc:175:50: error: 'pvs' was not declared in this scope
     out << " Number of G4VPhysicalVolume's: " << pvs->size() << std::endl;
                                                  ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/CondCore/Utilities/plugins/BasicP_PayloadInspector.cc 
>> Compiling LCG dictionary: tmp/slc6_amd64_gcc530/src/CondCore/Utilities/src/CondCoreUtilities/a/CondCoreUtilities_xr.cc
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/CondCore/CondDB/plugins/RelationalAuthenticationService.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/CondCore/CondDB/plugins/CondDBPyWrappers.cc 
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/FastSimulation/CaloGeometryTools/src/CrystalPad.cc:1:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/FastSimulation/CaloGeometryTools/interface/CrystalPad.h:26:49: error: use of undeclared identifier 'CLHEP'
  CrystalPad(unsigned number, const std::vector<CLHEP::Hep2Vector>& corners);
                                                ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-04-1100/src/FastSimulation/CaloGeometryTools/interface/CrystalPad.h:48:21: error: use of undeclared identifier 'CLHEP'
  bool inside(const CLHEP::Hep2Vector & point,bool debug=false) const;
                    ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/09ca96da5ef42a2e5f6970580b1194bd51158cb0
https://github.com/cms-sw/cmssw/commit/52ee4d0d0af1f56d49f7c29c1362d22825a80f13
https://github.com/cms-sw/cmssw/commit/d3aa85678e550cfd1f67138402a9638a50203890
https://github.com/cms-sw/cmssw/commit/6392f47e8d62121391f89e676d3a2635714b1741
https://github.com/cms-sw/cmssw/commit/fcd293dacbb35e50cee14eed39d823f1109ffed0
https://github.com/cms-sw/cmssw/commit/981939966af9100b414f344a153d94725044de41
https://github.com/cms-sw/cmssw/commit/c75f9f30c4fecf484e19874cb9dd7e30d997d554
https://github.com/cms-sw/cmssw/commit/b6eda340bd0a999942117f1d52da449a5ad28b95
https://github.com/cms-sw/cmssw/commit/62344a9088bdf3bcbaeb682005d1f1e7684de15e
https://github.com/cms-sw/cmssw/commit/06d1fe3cfefb629e50e980ebaae63a77b35ed7c7
https://github.com/cms-sw/cmssw/commit/03c43df9885cf604235c7563837928101145f473
https://github.com/cms-sw/cmssw/commit/e97d88ded6968a6c2fbcbcce57e1eaa6777e4c6a
https://github.com/cms-sw/cmssw/commit/c7ced6ae014d190ad85e440c12ff3f44f77323f1
https://github.com/cms-sw/cmssw/commit/66153ad310d01be1d512ceb632728cd0191b310a
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18211/18935/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18211/18935/git-merge-result

--- a/CondTools/SiPixel/test/SiPixelDynamicInefficiencyReader.cc
+++ b/CondTools/SiPixel/test/SiPixelDynamicInefficiencyReader.cc
@@ -11,6 +11,7 @@
 #include "DataFormats/SiPixelDetId/interface/PixelBarrelName.h"
 #include "DataFormats/SiPixelDetId/interface/PixelEndcapName.h"
 #include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 
 #include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
 

--- a/DataFormats/SiPixelDetId/interface/PixelBarrelName.h
+++ b/DataFormats/SiPixelDetId/interface/PixelBarrelName.h
@@ -7,11 +7,11 @@
 
 #include "DataFormats/SiPixelDetId/interface/PixelModuleName.h"
 #include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
-#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 
 #include <string>
 
-class DetId; 
+class DetId;
+class TrackerTopology;
 
 class PixelBarrelName : public PixelModuleName {
 public:

--- a/DataFormats/SiPixelDetId/interface/PixelEndcapName.h
+++ b/DataFormats/SiPixelDetId/interface/PixelEndcapName.h
@@ -5,13 +5,13 @@
  * Endcap Module name (as in PixelDatabase) for endcaps
  */
 #include "DataFormats/SiPixelDetId/interface/PixelModuleName.h"
-#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 #include "DataFormats/SiPixelDetId/interface/PXFDetId.h"
 
 #include <string>
 #include <iostream>
 
 class DetId;
+class TrackerTopology;
 
 class PixelEndcapName : public PixelModuleName {
 public:

--- a/DataFormats/SiPixelDetId/src/PixelBarrelName.cc
+++ b/DataFormats/SiPixelDetId/src/PixelBarrelName.cc
@@ -3,7 +3,7 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 //#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
-//#include "DataFormats/TarckerCommon/interface/TarckerTopology.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 
 #include <sstream>
 #include <iostream>

--- a/DataFormats/SiPixelDetId/src/PixelEndcapName.cc
+++ b/DataFormats/SiPixelDetId/src/PixelEndcapName.cc
@@ -2,6 +2,7 @@
 
 #include <sstream>
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 
 using namespace std;
 

